brokers/che-plugin-broker/broker/broker.go

```diff
@@ -0,0 +1,209 @@
+//
+// Copyright (c) 2012-2018 Red Hat, Inc.
+// This program and the accompanying materials are made
+// available under the terms of the Eclipse Public License 2.0
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+
+package broker
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+
+	"gopkg.in/yaml.v2"
+
+	"github.com/eclipse/che-go-jsonrpc"
+	"github.com/eclipse/che-plugin-broker/common"
+	"github.com/eclipse/che-plugin-broker/files"
+	"github.com/eclipse/che-plugin-broker/model"
+	"github.com/eclipse/che-plugin-broker/storage"
+)
+
+const pluginFileName = "che-plugin.yaml"
+const depFileName = "che-dependency.yaml"
+const depFileBothLocationAndURLError = "Plugin dependency '%s:%s' contains both 'location' and 'url' fields while just one should be present"
+const depFileNoLocationURLError = "Plugin dependency '%s:%s' contains neither 'location' nor 'url' field"
+
+// ChePluginBroker is used to process Che plugins
+type ChePluginBroker struct {
+	common.Broker
+	ioUtil  files.IoUtil
```

Comment:
what is the reason to have `ioUtil` as a field?

Reply:
To mock it. It is difficult to test behavior when implementation hides everything.